File: Estructura de Datos/3.cpp
#include <iostream>
#include <math.h>
#include <stdlib.h>  

using namespace std;

int main(){
	int lengthMatriz, maxValue, minValue;
	int arrCounter[10];
	
	cout << "Ingrese el numero de elementos de la matriz: ";
	cin >> lengthMatriz;
	/*cout << "Ingrese el valor maximo: ";
	cin >> maxValue;
	cout << "Ingrese el valor minimo: " ;
	cin >> minValue;
	*/
	
	int matriz[lengthMatriz];
	
	for(int i = 0; i < lengthMatriz; i++){
		matriz[i] = rand() % 10 + 1;
	}
	
	for(int i = 0; i < lengthMatriz; i++){
		int currentValue = matriz[i];
		int currentCounter = 0;
		for(int f = 0;f < lengthMatriz; f++){
			if(currentValue == matriz[f]){
				currentCounter++;
			}
		}
		cout << "'"<<currentValue<<"'"<< " tiene " <<currentCounter<<" coincidencias."<<endl;
	}
	
	//Mostrar los datos
	for(int i = 0; i < lengthMatriz; i++){
		cout << matriz[i]<<endl;
	}
	
	
	return 0;
}

File: Estructura de Datos/3.py
import random
# Realizar un código para averiguar
# cuantos números se repiten

# Variables globales
matriz = []
matrizFiltered = []
lengthMatriz = int(input('Ingrese la cantidad de numeros a generar: '))
maxValue = int(input('Ingrese el valor maximo:'))
minValue = int(input('Ingrese el valor minimo:'))

for i in range(lengthMatriz):
    matriz.append(random.randint(minValue, maxValue))
print('Matriz llenada!')
print(matriz)

for i in range(len(matriz)):
    currentValue = int(matriz[i])
    currentCounter = 0
    if (matrizFiltered.count(currentValue) != 1):
        for f in range(len(matriz)):
            currentCounter += (0, 1)[matriz[f] == currentValue]
        matrizFiltered.append(currentValue)
        print('[', currentValue, '] tiene ', currentCounter, ' coincidencias.')
